### PR TITLE
feat: new methods that return SendTransaction

### PR DESCRIPTION
**Pull request by r4mmer**
### Acceptance Criteria
- Create new methods that build transactions but return the `SendTransaction` instance instead of automatically mining the transactions.

### Security Checklist
- [ ] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/new/wallet.js [hunk 1]
```diff
@@ -1562,7 +1660,7 @@ class HathorWallet extends EventEmitter {
    * @param {boolean} [options.signTx] sign transaction instance (default true)
    * @param {boolean} [options.isCreateNFT=false] if the create token is an NFT creation call
    *
-   * @return {CreateTokenTransaction} Promise that resolves with transaction object if succeeds
+   * @return {Promise<CreateTokenTransaction>} Promise that resolves with transaction object if succeeds
```
Comment (pedroferreira1) [inline src/new/wallet.js:1663]: You could use in the docstring above the typedef you created for `CreateTokenOptions`

File: src/new/wallet.js [hunk 2]
```diff
@@ -1634,25 +1732,35 @@ class HathorWallet extends EventEmitter {
...
-   * @property {*} _dataToSignCache
-   */
-
-  /**
-   * @typedef CreateNewTokenResponse
-   * @extends BaseTransactionResponse
-   * @property {string} name
-   * @property {string} symbol
-   */
+   * Create a new token for this wallet
```
Comment (pedroferreira1) [inline src/new/wallet.js:1735]: Actually you create the send transaction object for a new token creation

File: src/new/wallet.js [hunk 3]
```diff
@@ -1634,25 +1732,35 @@ class HathorWallet extends EventEmitter {
...
+   *                                   Optional but required if not set in this
+   * @param {boolean} [options.createMint=true] should create mint authority
+   * @param {string} [options.mintAuthorityAddress] the address to send the mint authority created
+   * @param {boolean} [options.allowExternalMintAuthorityAddress=false] allow the mint authority address
+   *                                                                    to be from another wallet
+   * @param {boolean} [options.createMelt=true] should create melt authority
+   * @param {string} [options.meltAuthorityAddress] the address to send the melt authority created
+   * @param {boolean} [options.allowExternalMeltAuthorityAddress=false] allow the melt authority address
+   *                                                                    to be from another wallet
+   * @param {string[]|null} [options.data=null] list of data strings using utf8 encoding to add each as a data script output
```
Comment (pedroferreira1) [inline src/new/wallet.js:1754]: Why don't you use the `CreateTokenOptions` you created?

File: src/new/wallet.js [hunk 4]
```diff
@@ -1634,25 +1732,35 @@ class HathorWallet extends EventEmitter {
...
+   *
+   * @return {Promise<SendTransaction>}
+   * @memberof HathorWallet
+   * @inner
+   * */
+  async createNewTokenSendTransaction(name, symbol, amount, options = {}) {
+    const transaction = await this.prepareCreateNewToken(name, symbol, amount, options);
+    return new SendTransaction({ storage: this.storage, transaction });
```
Comment (pedroferreira1) [inline src/new/wallet.js:1762]: This line is throwing a warning that was not covered by tests but I thought we already had tests covering token creation, is that correct?

File: src/new/wallet.js [hunk 5]
```diff
@@ -1676,13 +1784,13 @@ class HathorWallet extends EventEmitter {
    *                                                                    to be from another wallet
    * @param {string[]|null} [options.data=null] list of data strings using utf8 encoding to add each as a data script output
```
Comment (pedroferreira1) [inline src/new/wallet.js:1785]: Why don't you use the `CreateTokenOptions` you created?

File: src/new/wallet.js [hunk 6]
```diff
@@ -1749,27 +1857,29 @@ class HathorWallet extends EventEmitter {
...
+   *                                   (if not sent we choose the next available address to use)
+   * @property {string} [changeAddress] address of the change output
+   *                                   (if not sent we choose the next available address to use)
+   * @property {boolean} [startMiningTx=true] boolean to trigger start mining (default true)
+   * @property {boolean} [createAnotherMint] boolean to create another mint authority or not for the wallet
+   * @property {string} [mintAuthorityAddress] address to send the new mint authority created
+   * @property {boolean} [allowExternalMintAuthorityAddress=false] allow the mint authority address to be from another wallet
+   * @property {boolean} [unshiftData] whether to unshift the data script output
+   * @property {string[]|null} [data=null] list of data strings using utf8 encoding to add each as a data script output
+   * @property {string} [pinCode] pin to decrypt xpriv information.
```
Comment (pedroferreira1) [inline src/new/wallet.js:1872]: Should we add `?` in all property types here?

File: src/new/wallet.js [hunk 7]
```diff
@@ -1749,27 +1857,29 @@ class HathorWallet extends EventEmitter {
...
+   */
```
Comment (pedroferreira1) [inline src/new/wallet.js:1873]: I think it's missing the `signTx` option, isn't it?

File: src/new/wallet.js [hunk 8]
```diff
@@ -1944,50 +2059,57 @@ class HathorWallet extends EventEmitter {
...
+   * @property {boolean} [options.startMiningTx=true] boolean to trigger start mining.
+   * @property {string} [options.pinCode] pin to decrypt xpriv information.
+   */
+
   /**
    * Prepare delegate authority transaction before mining
    *
    * @param {string} tokenUid UID of the token to delegate the authority
-   * @param {string} type Type of the authority to delegate 'mint' or 'melt'
+   * @param {string} type Type of the authority to delegate mint' or 'melt'
```
Comment (pedroferreira1) [inline src/new/wallet.js:2108]: Is it correct? I prefer without this change

Comment (r4mmer) [reply src/new/wallet.js]: Changed

File: src/new/wallet.js (hunk 7 above)
Comment (r4mmer) [reply src/new/wallet.js:1830]: Yes, added

File: src/new/wallet.js (hunk 6 above)
Comment (r4mmer) [reply src/new/wallet.js]: Done

File: src/new/wallet.js (hunk 5 above)
Comment (r4mmer) [reply src/new/wallet.js]: Nice catch!

File: src/new/wallet.js (hunk 3 above)
Comment (r4mmer) [reply src/new/wallet.js]: Nice catch!

File: src/new/wallet.js (hunk 2 above)
Comment (r4mmer) [reply src/new/wallet.js]: Changed

**Review by pedroferreira1 — APPROVED** [review]

**Review by andreabadesso — APPROVED** [review]

**Pull request merged**